internal/frontend/path_eval.go

```diff
@@ -246,6 +246,27 @@ func (eval *Eval) pathFunction(ctx context.Context, l *lua.State) (nResults int,
 	return 1, nil
 }
 
+func (eval *Eval) readFileFunction(ctx context.Context, l *lua.State) (int, error) {
+	path, err := lua.CheckString(l, 1)
+	if err != nil {
+		return 0, err
+	}
+	pcontext := l.StringContext(1)
```

Comment:
Oh whoops, one other thing: if the string context is non-empty, `readFile` should block on building the dependencies. Look at the `import` function implementation for an example.

Reply:
Ah, thanks!

(Only tangentially related: in what case would a string have a `len(pcontext) > 1`? Even if the string is a `derivation` return value, the string context itself would only be the store path, rather than the entire dependency chain of said derivation.

Reply:
In the case of `readFile`, `len(pcontext) > 1` would almost certainly be a user error.

`len(pcontext) > 1` is much more common for builder arguments and environment variables. Imagine some user code like:

```lua
derivation {
  -- ...
  builder = '/bin/sh';
  args = { '-c', "echo 'path1="..drv1.." path2="..drv2..'" };
}
```

The second argument to the builder would have two dependencies: one on `drv1` and one on `drv2`. Such a construct would almost certainly result in an invalid path for `readFile`.